Merge disruption

antoine-de opened the pull request:
Handling of Chaos disruptions in navitia.

still some work to be done (especially in jormungandr and we need to test the AT connector)

branch from #634 merged into dev

antoine-de committed:
Disruptions: objets serialization

azime committed:
nav2rt: load disruptions

antoine-de committed:
Disruption: change old disruption load

antoine-de committed:
Disruption: handle period for at messages

antoine-de committed:
Disruption: remove messages from PtObjects

Add Impacts instead of messages.

unit tests still have to be updated

antoine-de committed:
Disruption: refactor old tests

antoine-de committed:
Disruption: hide impacts and delete old classes

antoine-de committed:
Disruption: move at_perturbation from type to ed

It is used only in ed now (for adapted), so we moved at_perturbation to
ed

antoine-de committed:
Disruption: take comment into account

antoine-de committed:
Adapted: corrections after review

antoine-de committed:
Disruption: move a function to utils

antoine-de committed:
Disruption: refactor raptor adapted test

antoine-de committed:
Disruption: correction of mistake

TeXitoi committed:
kraken: cleanup DataManager

TeXitoi committed:
kraken: throw if DataManager::set_data have a nullptr

TeXitoi committed:
kraken: add forgotten include

TeXitoi committed:
kraken: importing disruption in Data when Chaos send one to us

antoine-de committed:
Disruption: make test buildable

TeXitoi committed:
kraken: add utils in library of data_manager_test

TeXitoi committed:
ed, krakenk type: migrate to boost::shared_ptr

TeXitoi committed:
kraken: move filling disruption in its own file

TeXitoi committed:
type, ed, kraken: add impacts in pt_objs

antoine-de committed:
Disruption: refacto disruption test

antoine-de committed:
Disruption: change after review and test correction

antoine-de committed:
Kraken: fix shared_ptr type after merge

antoine-de committed:
Kraken: add cast for old boost version

kinnou02 commented on source/type/realtime.proto:
```diff
@@ -6,30 +6,6 @@ message TCObject{
     optional pbnavitia.NavitiaType object_type = 2;
 }
 
-message LocalizedMessage{
```
it's used in sindri and at_connector no?

antoine-de replied:
doesn't seems no (nothing with git grep LocalizedMessage)

kinnou02 replied:
sindri/sindri/saver/message.py:126
connectors/connectors/at/selector/atreader.py:253

antoine-de committed:
Disruption: put back old messages

for the transition, we keep old messages in pt object to dupplicate the
disruptions.

as soon as NMP (our only customer using disruptions) migrate to the new
version, we'll drop the messages

antoine-de committed:
All: add utilities test methods for date objects

enable date creation from a string "20140101"_d

antoine-de committed:
Disruptions: preparation for the integration tests

l-vincent-l committed:
Kraken: add chaos configuration

antoine-de committed:
Disruptions: add integration tests on /disruptions

antoine-de committed:
Jormungandr: check size after loop to check the elts

l-vincent-l committed:
Kraken: load disruptions from database

antoine-de committed:
Tests: add cli argument to mock kraken

antoine-de committed:
Jormun: add test with mock chaos

l-vincent-l committed:
Kraken: fill channels from database

l-vincent-l committed:
Kraken: make fill_disruption_from_database testable

l-vincent-l committed:
Kraken: remove useless includes

antoine-de committed:
Disruption: corrections after review

antoine-de committed:
Disruption: add a last_rt_data_loaded field

to be able to now the date of the last reload of data due to rt, add a
field with the datetime of the last rt data integration.

the field is not serialized (I don't see the point)

l-vincent-l committed:
Kraken: some lint

l-vincent-l committed:
Kraken: add test fill_disruption_database

antoine-de committed:
Test: remove boost::any_of since it's not in boost 1.49

antoine-de committed:
Jormungandr: correction of auto pycharm import...

l-vincent-l committed:
Kraken: add read_one_line test

antoine-de committed:
Disruption: compute disruption status

antoine-de committed:
Jormungandr: add one test on disruptions

l-vincent-l committed:
Kraken: add test on fill_disruption_from_database

l-vincent-l committed:
Kraken: fix bugs in fill_disruption_from_database

l-vincent-l committed:
Kraken: use boost optional

l-vincent-l committed:
Kraken: add a nice exception message

l-vincent-l committed:
Kraken: add comment

l-vincent-l committed:
Kraken: fix init of chaos_database

l-vincent-l committed:
Kraken: use unique_ptr on full_disruptiion_database's disruption object

l-vincent-l committed:
Kraken: use boost_lexical_cast in tests

l-vincent-l committed:
Kraken: add default value for chaos_database

antoine-de committed:
Jormungandr: generate dyn rt topic and wait for cnx

we need to wait for the rabbitmq connection

antoine-de committed:
Jormungandr: change poll to library retrying

antoine-de committed:
Disruption: change typo on 'futur'

antoine-de committed:
Jormungandr: fix typo

antoine-de committed:
Jormungandr: change timeout on rabbitmq cnx poll

antoine-de committed:
Jormungandr: remove unsused var

antoine-de committed:
Jormungandr: correction for old protobuf version

TeXitoi commented on source/disruption/tests/disruption_test.cpp:
```diff
@@ -57,97 +59,155 @@ start_application_daily_hour = 08h40
+using navitia::type::new_disruption::Disruption;
+
+enum class ChaosType {
+    //we copy the chaos type because we don't want chaos dependencies here
```
Why we don't want chaos dependencies in a test? who care?

TeXitoi commented on source/ed/at_perturbation.cpp:
```diff
@@ -0,0 +1,87 @@
+        return false;
+    }
+
+    //on doit travailler jour par jour
```
french comments? ;)

TeXitoi commented on source/ed/connectors/messages_connector.cpp:
```diff
@@ -35,12 +35,47 @@ www.navitia.io
+        if (auto res = it->second.lock()) return res;
+    }
+
+    boost::shared_ptr<Severity> severity = boost::make_shared<Severity>();
```
auto is great for useless repetition of type like this

TeXitoi commented on source/ed/connectors/messages_connector.cpp:
```diff
@@ -69,114 +104,71 @@ std::map<std::string, boost::shared_ptr<navitia::type::Message>> load_messages(
-            message->object_type = static_cast<navitia::type::Type_e>(
-                    cursor["object_type_id"].as<int>());
+    for (auto cursor = result.begin(); cursor != result.end(); ++cursor) {
+        //we can have several language handled by the database, so we create one message for each
```
I don't understand this comment. where is the loop adding a message for each language?

kinnou02 commented on source/ed/connectors/messages_connector.cpp:
```diff
@@ -35,12 +35,47 @@ www.navitia.io
+    disruptions.severities.insert({name, severity});
+
+    severity->uri = name;
+    severity->effect = Effect::UNKNOWN_EFFECT; //is it right ?
```
yes

kinnou02 commented on source/kraken/fill_disruption_from_chaos.cpp:
```diff
@@ -0,0 +1,211 @@
+    impact->severity = make_severity(chaos_impact.severity(), holder);
+    impact->informed_entities = make_pt_objects(chaos_impact.informed_entities(), pt_data, impact);
+    for (const auto& chaos_message: chaos_impact.messages()) {
+        impact->messages.push_back({
```
I don't see where we handle the channel

antoine-de replied:
nop, they're not handled for the moment.
we just need to read them or do we need some filters to ?

kinnou02 commented on source/kraken/maintenance_worker.cpp:
```diff
@@ -94,15 +99,22 @@ void MaintenanceWorker::handle_rt(AmqpClient::Envelope::ptr_t envelope){
-            data_manager.apply_disruptions(0);
+            if (!data) {
+                data = data_manager.get_data_clone();
+                data->last_rt_data_loaded = pt::microsec_clock::local_time();
```
local or utc?

antoine-de committed:
Disruption: use chaos type in test

antoine-de committed:
Disruptions: misc corrections after reviews

TeXitoi commented:
OK for me.

antoine-de committed:
Disruption: correction after review
